scripts/git/update_to_llvm_syncpoint.py

```diff
@@ -0,0 +1,248 @@
+      type=utils.str2bool,
+      nargs="?",
+      default=True,
+  )
+
+  parser.add_argument("--update_build_files",
+                      help="Updates the IREE LLVM build files from TensorFlow.",
+                      type=utils.str2bool,
+                      nargs="?",
+                      default=True)
+  args = parser.parse_args()
+
+  # Default repo path.
+  if args.repo is None:
+    args.repo = utils.find_git_toplevel()
+
+  # Set some defaults.
+  if not args.tensorflow:
+    args.tensorflow = os.path.join(args.repo, "third_party", "tensorflow")
+  if not args.llvm:
+    args.llvm = os.path.join(args.repo, "third_party", "llvm-project")
+
+  return args
+
+
+def main(args):
+  print("IREE handy-dandy-LLVM-submodule-updater at your service...")
+  print("  IREE Path :", args.repo)
+  print("  LLVM Path :", args.llvm)
+  print("  TensorFlow Path :", args.tensorflow)
+  print("  Update Build files:", args.update_build_files)
+  current_llvm_commit = get_commit(args.llvm)
+  current_tensorflow_commit = get_commit(args.tensorflow)
+
+  print("Current Commits: llvm =", current_llvm_commit, "tensorflow =",
+        current_tensorflow_commit)
+
+  # Update TensorFlow
+  new_tf_commit = find_new_tf_commit(args.tensorflow, current_llvm_commit,
+                                     args.tensorflow_commit)
+  print("\n*** Updating TensorFlow to", new_tf_commit, "***")
+  utils.execute(["git", "checkout", new_tf_commit], cwd=args.tensorflow)
+  stage_path(args.repo, args.tensorflow)
+
+  validate_tf_commit(current_llvm_commit,
+                     args.tensorflow,
+                     exit_on_failure=args.validate)
+
+  if args.update_build_files:
+    print("\n*** Updating BUILD.bazel files ***")
+    update_build_files_from_tensorflow(args.repo, args.tensorflow)
+
+  # Export SUBMODULE_VERSIONS.
+  print()  # Add line break.
+  submodule_versions.export_versions(args.repo)
+
+
+def get_commit(path, rev="HEAD"):
+  return utils.execute(["git", "rev-parse", rev],
+                       cwd=path,
+                       silent=True,
+                       capture_output=True,
+                       universal_newlines=True).strip()
+
+
+def find_new_tf_commit(tensorflow_path, llvm_commit, tf_commit):
+  utils.execute(["git", "fetch"], cwd=tensorflow_path)
+
+  if tf_commit not in COMMIT_OPTIONS:
+    return get_commit(tensorflow_path, rev=tf_commit)
+
+  if tf_commit == KEEP_COMMIT:
+    return get_commit(tensorflow_path)
+
+  if tf_commit == REMOTE_HEAD_COMMIT:
+    return get_commit(tensorflow_path, "origin/master")
+
+  tf_integrate_commits = utils.execute([
+      "git", "log", "--first-parent", "--format=%H", "-S", llvm_commit,
+      "origin/master", "--", "tensorflow/workspace.bzl"
+  ],
+                                       capture_output=True,
+                                       universal_newlines=True,
+                                       cwd=tensorflow_path).split()
```

Comment:
lol @ the whitespace here. Wonder if our python autoformatting will adjust it :P

Reply:
I'm trying to use the same yapf formatter as internally, but it does not appear to always work the same...